OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/resetOSPassword.sh

```diff
@@ -0,0 +1,214 @@
+#!/bin/bash
+# LICENSE UPL 1.0
+#
+# Copyright (c) 1982-2018 Oracle and/or its affiliates. All rights reserved.
+# 
+# Since: January, 2018
+# Author: paramdeep.saini@oracle.com
+# Description: Reset the password for Grid and oracle user
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+# 
+
+declare -a cluster_nodes
+GRID_USER='grid'
+ORACLE_USER='oracle'
+PROGNAME=$(basename $0)
+PWD_KEY='pwd.key'
+SECRET_VOLUME='/run/secrets'
+declare -x REMOVE_OS_PWD_FILES='false'
+
+source /etc/rac_env_vars
+
+###################Capture Process id and source functions.sh###############
+source "$SCRIPT_DIR/functions.sh"
+###########################sourcing of functions.sh ends here##############
+
+####error_exit function sends a TERM signal, which is caught by trap command and returns exit status 15"####
+trap '{ exit 15; }' TERM
+###########################trap code ends here##########################
+
+generate_pwd ()
+{
+
+if [ -f "${SECRET_VOLUME}/${PWD_FILE}" ] && [ -f "${SECRET_VOLUME}/${PWD_KEY}" ] ; then
+cmd='openssl enc -d -aes-256-cbc -in "${SECRET_VOLUME}/${PWD_FILE}" -out /tmp/${PWD_FILE} -pass file:"${SECRET_VOLUME}/${PWD_KEY}"'
+
+eval $cmd
+
+if [ $? -eq 0 ]; then
+print_message "Password file generated"
+else
+error_exit "Error occurred during common os password file generation"
+fi
+
+read PASSWORD < /tmp/${PWD_FILE}
+rm -f /tmp/${PWD_FILE}
+rm -f /tmp/${PWD_KEY}
+
+if [ "${REMOVE_OS_PWD_FILES}" == 'true' ]; then
+rm -f  ${SECRET_VOLUME}/${COMMON_OS_PWD_FILE}
+rm -f ${SECRET_VOLUME}/${PWD_KEY}
+fi
+
+else
+ print_message "Password file or password key file is empty string! generating random password"
+ PASSWORD=O$(openssl rand -base64 6 | tr -d "=+/")_1
+fi
+
+echo  $PASSWORD
```

Comment:
This echo should be commented out or only enabled when debugging

Reply:
Fixed it.